### ISSUE TITLE
ansible-galaxy won't show it's version if called without an action

### ISSUE THREAD

**Issue by muffl0n** [ansible/ansible#12004]
##### Issue Type:

Bug Report
##### Ansible Version:

2.0.0 (devel 1b810e3101)
##### Ansible Configuration:

n/a
##### Environment:

Gentoo Linux, Python 2.7.9
##### Summary:

To see the version of ansible-galaxy one has to provide a fake action because without an action ansible-galaxy will just show the usage and an error message.
##### Steps To Reproduce:

Clone current devel, switch to hacking and execute `ansible-galaxy --version`
##### Expected Results:

```
$ ansible-galaxy --version
ansible-galaxy 2.0.0 (devel 1b810e3101) last updated 2015/08/19 12:39:34 (GMT +200)
  lib/ansible/modules/core: (detached HEAD dea5f5d15e) last updated 2015/08/19 12:40:20 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 210ee3febc) last updated 2015/08/19 12:40:33 (GMT +200)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/08/19 12:40:47 (GMT +200)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/08/19 12:41:00 (GMT +200)
  config file = 
  configured module search path = None
```
##### Actual Results:

```
$ ansible-galaxy --version
Usage: ansible-galaxy [init|info|install|list|remove] [--help] [options] ...

Options:
  -h, --help     show this help message and exit
  -v, --verbose  verbose mode (-vvv for more, -vvvv to enable connection
                 debugging)
  --version      show program's version number and exit
ERROR! Missing required action
```

**Comment by muffl0n** [comment]
The problem seems to be that `self.set_action()` in `GalaxyCLI.parse()` (/lib/ansible/cli/galaxy.py) is called before optparse can parse the arg and show the version infos. `CLI.set_action()` then throws `AnsibleOptionsError("Missing required action")` because we obviously did not specify an action.

I found that calling `self.parser.parse_args()` before `self.set_action()` solves this problem but results in calling this method twice: We are adding several options to optparse after determining the action and therefore have to call `self.parser.parse_args()` again after that.

I'm not quite sure if this is the best solution

**Comment by bcoca** [comment]
ansible-vault will probably have the same issue

**Comment by muffl0n** [comment]
I submitted a PR for my proposed fix: #12020. 

My original plan was to get rid of the manual setting of the usage e.g. in vault.py:

```
        # options specific to self.actions
        if self.action == "create":
            self.parser.set_usage("usage: %prog create [options] file_name")
```

So one would not need this workaround. But unfortunately optparse does not support positional arguments. :(

**Comment by gildegoma** [comment]
I'd like to add my two cents to urge Ansible core team to consider fixing this for the final release of Ansible 2.0.0, announced for January 12 (e.g. by merging #12020, which works fine for me).

In addition to not displaying the requested version, `ansible-galaxy --version` and `ansible-vault --version` calls are then returning a `5` code instead of the expected `0` (success). Without any fix, some software that integrate these Ansible CLI tools will possible blow up :dash: (e.g. Vagrant Ansible provisioners). For this reason, I am pretty sure it is worth to fix it "asap" :)

**Comment by bcoca** [comment]
we were not prioritizing this yet as calling `ansible --version` would give you the output you need as they all share the same version (they are actually symlinks to bin/ansible in 2.0).

**Comment by gildegoma** [comment]
@bcoca thanks for the very quick feedback. Good to know that we have to count with this issue in the early days of Ansible 2+ ;-)

**Comment by bcoca** [comment]
fixed via manual merge of #12020 via 11b55be5bbb90b2bc917b2637d6fcdbe1a15092d

**Comment by habnabit** [comment]
This fix has spawned another problem: #13743

**Comment by bcoca** [comment]
reverted merge

**Comment by muffl0n** [comment]
I opened #13762 to address this issue. I'm now explicitly checking for "--version". Not as clean as I wished it to be, but I lack an idea for a better solution.

**Comment by bcoca** [comment]
fixed via e63bf1d86e9e54662781cb21ab0e65508cf51759
